Temporary folder is never cleaned up

**Describe the bug**
The temporary and logs folder never get cleaned out, slowly consuming the drive where wav2bar is installed.

**To Reproduce**
Run a number of renders.
Check the temporary folder and logs folder.

**Expected behavior**
Temporary files are cleared after each render.
Oldest logs get removed if they grow too large.

**Additional Info**
My system showed over 100gb of usage in the wav2bar temp folder and 2.3 around logs.
A friend showed over 60gb of usage in the wav2bar temp folders.

Windows:
* temp folder is under the wav2bar folder in /tmp (eg: c:\wav2bar\tmp)
* log folder is under the wav2bar folder in /logs (eg: c:\wav2bar\logs)

Mac/OSX: 
I ran the build commands to get wav2bar running:
* temp folder: ~/wav2bar/out/Wav2Bar-darwin-arm64/Wav2Bar.app/Contents/Resources/app/temp/
* logs folder:  ~/wav2bar/out/Wav2Bar-darwin-arm64/Wav2Bar.app/Contents/Resources/app/logs



I cleaned out the temp/renders folder via: find . -delete (while prompt is in the renders folder)
I used find instead of rm due to an "argument list too long" error using rm.
I cleaned out the logs/ folder via: rm * (while prompt is in the logs folder)

Some limited feedback after a render on OSX.

I get a small error during shutdown:
```
[2024-06-06T13:20:00.824] [INFO] main_renderer - The window will be closed.
[2024-06-06T13:20:00.825] [ERROR] main_renderer - Uncaught TypeError: Cannot read properties of null (reading 'export_mode')
source: file:///Users/makkura/wav2bar/out/Wav2Bar-darwin-arm64/Wav2Bar.app/Contents/Resources/app/js/index.js
 line: 1504
[2024-06-06T13:20:00.844] [INFO] main - closing main renderer... 
```

After shut down, my render folder was 82GB. So part of the issue is just a large operational space.
After starting wav2bar again, the render folder was cleaned up.

Will also test doing 2 renders without closing the program and checking folder size.
I only noticed this due to running out of space on my mac so could be an issue with rendering without closing between or just a larger render than others.


No update on Windows as of yet. My friend is rendering on windows so I have to get that info second hand.

--- a/index.html
+++ b/index.html
@@ -20,10 +20,11 @@ along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
 <!DOCTYPE html>
 <html lang="en">
+
 <head>
-    <meta charset="UTF-8"/>
-    <meta name="viewport" content="width=device-width, initial-scale=1.0"/>
-    <meta http-equiv="X-UA-Compatible" content="ie=edge"/>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <meta http-equiv="X-UA-Compatible" content="ie=edge" />
     <!-- <meta http-equiv="Content-Security-Policy" content="default-src 'self';"/> -->
     <title>Wav2Bar</title>
 
@@ -37,8 +38,8 @@ along with this program.  If not, see <https://www.gnu.org/licenses/>.
     <script type="text/javascript" src="js/export_main.js"></script>
 
     <!--CSS-->
-    <link rel="stylesheet" type="text/css" href="style.css"/>
-    <link rel="stylesheet" type="text/css" href="node_modules/remixicon/fonts/remixicon.css"/>
+    <link rel="stylesheet" type="text/css" href="style.css" />
+    <link rel="stylesheet" type="text/css" href="node_modules/remixicon/fonts/remixicon.css" />
 </head>
 
 
@@ -47,7 +48,7 @@ along with this program.  If not, see <https://www.gnu.org/licenses/>.
         <ui-split-layout data-direction="vertical" data-size-min="300" data-size-max="600">
             <!--CONTROL PANEL-->
             <div slot="first-child" id="control_panel">
-    
+
                 <!--Labels, tabs naviguation-->
                 <div id="naviguation_tab_labels">
                     <div class="tab_label" id="project_label"> <i class="ri-file-edit-line"></i></i> PROJECT</div>
@@ -56,26 +57,21 @@ along with this program.  If not, see <https://www.gnu.org/licenses/>.
                     <div class="tab_label" id="settings_label"> <i class="ri-settings-4-line"></i> SETTINGS</div>
                     <div class="tab_label" id="help_label"> <i class="ri-question-line"></i> CONTACT AND HELP</div>
                 </div>
-    
-    
-    
+
+
+
                 <!--Tabs container-->
                 <div id="tabs_container">
-    
+
                     <!--PROJECT-->
                     <div id="project_tab" class="tab">
 
                         <ui-parameter data-help="fps">
-                            <ui-input-field id="fps-input"
-                                data-name="Framerate"
-                                data-type="int"
-                                data-value="0"
-                                data-min="1"
-                                data-max="8000"
-                                data-step="1">
+                            <ui-input-field id="fps-input" data-name="Framerate" data-type="int" data-value="0"
+                                data-min="1" data-max="8000" data-step="1">
                             </ui-input-field>
                         </ui-parameter>
-    
+
                         <!-- <div class="panel_param_container" data-help="screen_size">
                             <ui-label-block data-name="Screen resolution">
                                 <div>
@@ -87,51 +83,46 @@ along with this program.  If not, see <https://www.gnu.org/licenses/>.
                         </div> -->
 
                         <ui-parameter data-help="screen_size" data-title="Screen resolution">
-                            <ui-input-field id="screen-width-input"
-                                data-name="Width"
-                                data-type="int"
-                                data-value="1280"
-                                data-min="16"
-                                data-max="8000"
-                                data-step="1">
+                            <ui-input-field id="screen-width-input" data-name="Width" data-type="int" data-value="1280"
+                                data-min="16" data-max="8000" data-step="1">
                             </ui-input-field>
-                            <ui-input-field id="screen-height-input"
-                                data-name="Height"
-                                data-type="int"
-                                data-value="720"
-                                data-min="16"
-                                data-max="8000"
-                                data-step="1">
+                            <ui-input-field id="screen-height-input" data-name="Height" data-type="int" data-value="720"
+                                data-min="16" data-max="8000" data-step="1">
                             </ui-input-field>
                         </ui-parameter>
 
                         <ui-parameter id="load-audio" data-title="Import audio (mp3/wav/ogg)" data-help="audio">
-                            <ui-file-picker id="load-audio-picker" data-type="get_file" data-button-text="CHOOSE AUDIO FILE" data-opened-span-text="Opened audio file"></ui-file-picker>
+                            <ui-file-picker id="load-audio-picker" data-type="get_file"
+                                data-button-text="CHOOSE AUDIO FILE"
+                                data-opened-span-text="Opened audio file"></ui-file-picker>
                         </ui-parameter>
                         <!-- <div class="panel_param_container" data-help="audio">
                             <button id="load_audio_button" class="panel_button">CHOOSE AUDIO FILE</button>
                             <div><span>: </span><span id="opened_audio"></span></div>
                         </div> -->
-                        
+
                         <ui-parameter id="load-save" data-title="Import save (.w2bzip)" data-help="save_import">
-                            <ui-file-picker id="load-save-picker" data-type="get_file" data-button-text="CHOOSE SAVE FILE" data-opened-span-text="Opened save"></ui-file-picker>
+                            <ui-file-picker id="load-save-picker" data-type="get_file"
+                                data-button-text="CHOOSE SAVE FILE"
+                                data-opened-span-text="Opened save"></ui-file-picker>
                         </ui-parameter>
                         <!-- <div class="panel_param_container" data-help="save_import">
                             <span>Import save (.w2bzip): </span>
                             <button id="save_file_button" class="panel_button">CHOOSE SAVE FILE</button>
                             <div><span>Opened save: </span><span id="opened_save"></span></div>
                         </div> -->
-    
+
                         <ui-parameter id="export-save" data-title="Export save" data-help="save_export">
-                            <ui-file-picker id="export-save-picker" data-type="save_file" data-button-text="SAVE" data-opened-span-text="Saved at"></ui-file-picker>
+                            <ui-file-picker id="export-save-picker" data-type="save_file" data-button-text="SAVE"
+                                data-opened-span-text="Saved at"></ui-file-picker>
                         </ui-parameter>
                         <!-- <div class="panel_param_container" data-help="save_export">
                             <span>Export save: </span>
                             <button id="export_save_button" class="panel_button">SAVE</button>
                         </div> -->
-                    
+
                     </div>
-    
+
                     <!--OBJECTS-->
                     <div id="objects_tab" class="tab">
                         <ui-parameter data-help="new_object" data-title="New Object">
@@ -149,13 +140,13 @@ along with this program.  If not, see <https://www.gnu.org/licenses/>.
                                         <option value="visualizer_straight_wave">Visualizer - straight, waves</option>
                                         <option value="visualizer_circular_bar">Visualizer - circular</option>
                                     </optgroup>
-                                </select>    
+                                </select>
                             </ui-label-block>
                             <button id="create_object_button" class="panel_button panel_button_wide">CREATE</button>
                         </ui-parameter>
                         <!--all object parameters container go here-->
                     </div>
-    
+
                     <!--EXPORT-->
                     <div id="export_tab" class="tab">
                         <ui-parameter id="choose-video-path" data-title="Save directory" data-help="export_video_path">
@@ -166,17 +157,26 @@ along with this program.  If not, see <https://www.gnu.org/licenses/>.
                             <input id="video_export_path_input" class="panel_input panel_input_string" />
                             <button id="choose_video_path_button" class="panel_button">BROWSE</button>
                         </div> -->
-    
+
                         <!-- <div class="panel_param_container" data-help="experimental_jpeg_export">
                             <span>Experimental JPEG export: </span>
                             <input id="experimental_export_input" type="checkbox" class="panel_input panel_input_checkbox" />
                         </div> -->
                         <ui-parameter data-help="experimental_jpeg_export">
-                            <ui-input-field id="experimental-export-input"
-                                data-name="Use JPEG for frames export"
-                                data-type="bool"
-                                data-value="false">
+                            <ui-input-field id="experimental-export-input" data-name="Use JPEG for frames export"
+                                data-type="bool" data-value="false">
                             </ui-input-field>
+                        </ui-parameter>
+
+                        <ui-parameter>
+                            <p>
+                                Warning: due to how export is handled in Wav2Bar, export can take up a lot of storage
+                                space on disk (around 100GB). Make sure you have enough space on your disk before
+                                exporting.
+                            </p>
+                            <p>
+                                For more information, report to this issue: https://github.com/Picorims/wav2bar/issues/69
+                            </p>
                         </ui-parameter>
 
                         <ui-parameter data-title="Export" data-help="export">
@@ -191,7 +191,7 @@ along with this program.  If not, see <https://www.gnu.org/licenses/>.
                             <div class="progress_bar">
                                 <div id="export_frame_progress" class="progress_child"></div>
                             </div>
-                            
+
                             <ui-label-block data-name="Encoding video">
                                 <span id="export_encoding_span"></span>
                             </ui-label-block>
@@ -221,11 +221,12 @@ along with this program.  If not, see <https://www.gnu.org/licenses/>.
                             </div>
                         </div> -->
                     </div>
-    
+
                     <!--SETTINGS-->
                     <div id="settings_tab" class="tab">
                         <ui-parameter data-title="Logs">
-                            <button id="open_logs_button" class="panel_button panel_button_wide">BROWSE LOGS FOLDER</button>    
+                            <button id="open_logs_button" class="panel_button panel_button_wide">BROWSE LOGS
+                                FOLDER</button>
                         </ui-parameter>
                         <ui-parameter id="locate-ffmpeg" data-title="FFmpeg location">
                             <ui-file-picker id="locate-ffmpeg-picker" data-type="get_file"></ui-file-picker>
@@ -247,7 +248,7 @@ along with this program.  If not, see <https://www.gnu.org/licenses/>.
                             <button id="open_ffmpeg_help" class="panel_button panel_button_wide">OPEN HELP</button>
                         </ui-parameter>
                     </div>
-    
+
                     <!--CONTACT AND HELP-->
                     <div id="help_tab" class="tab">
                         <ui-parameter class="about_info">
@@ -255,75 +256,77 @@ along with this program.  If not, see <https://www.gnu.org/licenses/>.
                                 Wav2Bar v<span id="software_version"></span>
                             </p>
                             <p>
-                                Place your mouse over the <i class="ri-question-line"></i> icons to get help on a specific feature.
+                                Place your mouse over the <i class="ri-question-line"></i> icons to get help on a
+                                specific feature.
                             </p>
                             <p>
                                 Information and contact:
-                                <br/>
-                                <a id="website_link">Website</a><br/>
-                                <a id="github_link">GitHub repository</a><br/>
-                                <a id="twitter_link">Twitter</a><br/>
-                                <a id="youtube_link">YouTube</a><br/>
-                                <a id="discord_link">Discord Server</a><br/>
+                                <br />
+                                <a id="website_link">Website</a><br />
+                                <a id="github_link">GitHub repository</a><br />
+                                <a id="twitter_link">Twitter</a><br />
+                                <a id="youtube_link">YouTube</a><br />
+                                <a id="discord_link">Discord Server</a><br />
                             </p>
                             <p>
-                                Send an email at: <a href="mailto:picorims.contact@gmail.com">picorims.contact@gmail.com</a>
+                                Send an email at: <a
+                                    href="mailto:picorims.contact@gmail.com">picorims.contact@gmail.com</a>
                             </p>
                             <p>
                                 Special thanks:
-                                <br/>
-                                - My amazing friend Kanade_Senpai<br/>
+                                <br />
+                                - My amazing friend Kanade_Senpai<br />
                                 - RemixIcon (remixicon.com)
                             </p>
                             <hr>
                             <p>
                                 Wav2Bar - Free software for creating audio visualization (motion design) videos
-                                <br/>Copyright (C) 2023  Picorims &lt;picorims.contact@gmail.com&gt;
-                                <br/><br/>
+                                <br />Copyright (C) 2023 Picorims &lt;picorims.contact@gmail.com&gt;
+                                <br /><br />
                                 This program is free software: you can redistribute it and/or modify
                                 it under the terms of the GNU General Public License as published by
                                 the Free Software Foundation, either version 3 of the License, or
                                 any later version.
-                                <br/><br/>
+                                <br /><br />
                                 This program is distributed in the hope that it will be useful,
                                 but WITHOUT ANY WARRANTY; without even the implied warranty of
-                                MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+                                MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
                                 GNU General Public License for more details.
-                                <br/><br/>
+                                <br /><br />
                                 You should have received a copy of the GNU General Public License
-                                along with this program.  If not, see &lt;https://www.gnu.org/licenses/&gt;.
+                                along with this program. If not, see &lt;https://www.gnu.org/licenses/&gt;.
                             </p>
                             <p>
                                 Helpers:
-                                <br/>
-                                - Kanade_Senpai (trailer video rendering)<br/>
+                                <br />
+                                - Kanade_Senpai (trailer video rendering)<br />
                                 - pdejan (fix help.json information)
                             </p>
                         </ui-parameter>
                     </div>
                 </div>
             </div>
-    
-    
-    
-    
-    
+
+
+
+
+
             <!--INTERFACE WITH SCREEN-->
             <div slot="second-child" id="interface">
-                
+
                 <div id="screen_container">
                     <!-- PREVIEW / RENDER -->
                     <div id="screen">
                         <!-- objects goes here! -->
                     </div>
-        
+
                     <!-- INNER SPACING FIXER -->
                     <div id="inner_spacing_fixer"></div>
                 </div>
 
                 <!--Naviguation Buttons-->
                 <div id="top_UI">
-    
+
                     <!--Audio control-->
                     <div class="top_UI_block">
                         <button class="top_UI_tile left_round" id="play_audio">
@@ -339,34 +342,34 @@ along with this program.  If not, see <https://www.gnu.org/licenses/>.
                             <span class="text" id="time_display">0:00 | 0:00</span>
                         </div>
                     </div>
-    
+
                     <!--audio position-->
                     <div class="top_UI_block">
                         <button class="top_UI_tile left_round" id="audio_to_start">
                             <i class="ri-skip-back-fill"></i>
                         </button>
                         <div class="top_UI_tile range">
-                            <input class="range_track" type="range" id="audio_range"/>
+                            <input class="range_track" type="range" id="audio_range" />
                         </div>
                         <button class="top_UI_tile right_round" id="audio_to_end">
                             <i class="ri-skip-forward-fill"></i>
                         </button>
                     </div>
-    
+
                     <!--Loop music-->
                     <div class="top_UI_block">
                         <button class="top_UI_tile left_round right_round" id="loop_audio">
                             <i class="ri-repeat-fill"></i>
                         </button>
                     </div>
-    
+
                     <!--FPS-->
                     <div class="top_UI_block">
                         <div class="top_UI_tile left_round right_round double_size">
                             <span id="fps" class="text"></span>
                         </div>
                     </div>
-    
+
                     <!--Zoom-->
                     <div class="top_UI_block">
                         <button class="top_UI_tile left_round" id="zoom_out">
@@ -375,16 +378,16 @@ along with this program.  If not, see <https://www.gnu.org/licenses/>.
                         <button class="top_UI_tile double_size" id="zoom_value">
                             <span id="zoom_display" class="text"></span>
                         </button>
-                        <button class="top_UI_tile right_round"id="zoom_in">
+                        <button class="top_UI_tile right_round" id="zoom_in">
                             <i class="ri-zoom-in-line"></i>
                         </button>
                     </div>
                 </div>
-    
+
             </div>
         </ui-split-layout>
-      
-    
+
+
         <!--FILTERS TESTS-->
         <!--<svg xmlns="http://www.w3.org/2000/svg" version="1.1">
             <defs>
@@ -413,4 +416,5 @@ along with this program.  If not, see <https://www.gnu.org/licenses/>.
         </svg>-->
     </div>
 </body>
+
 </html>

--- a/index.html
+++ b/index.html
@@ -228,6 +228,10 @@ along with this program.  If not, see <https://www.gnu.org/licenses/>.
                             <button id="open_logs_button" class="panel_button panel_button_wide">BROWSE LOGS
                                 FOLDER</button>
                         </ui-parameter>
+                        <ui-parameter data-title="Temp">
+                            <button id="open_temp_button" class="panel_button panel_button_wide">BROWSE TEMP
+                                FOLDER</button>
+                        </ui-parameter>
                         <ui-parameter id="locate-ffmpeg" data-title="FFmpeg location">
                             <ui-file-picker id="locate-ffmpeg-picker" data-type="get_file"></ui-file-picker>
                         </ui-parameter>

--- a/js/export_main.js
+++ b/js/export_main.js
@@ -159,9 +159,10 @@ function Export(path) {
                 let secs = Math.floor((ellapsed/1000)%60);
                 mins = (mins<10)? "0"+mins : mins;
                 secs = (secs<10)? "0"+secs : secs;
-
+                
                 MessageDialog("info",`The video has been successfully created in ${hours}:${mins}:${secs} !`);
                 imports.utils.CustomLog("info",`The video has been successfully created in ${hours}:${mins}:${secs} !`);
+                ipcRenderer.invoke("clean-up-render-files");
             }
             else MessageDialog("error","An error occurred during the encoding process. For more information, see the logs.");
         });

--- a/js/user_interface.js
+++ b/js/user_interface.js
@@ -248,6 +248,11 @@ async function InitUI() {
         ipcRenderer.invoke("open-folder-in-file-explorer", `${project.working_dir}/logs`);
     };
 
+    //open logs folder
+    document.getElementById("open_temp_button").onclick = function() {
+        ipcRenderer.invoke("open-folder-in-file-explorer", `${project.working_dir}/temp`);
+    };
+
     //choose ffmpeg path through file browser
     /** @type {uiComponents.WebUIFilePicker} */
     let ffmpeg_picker = document.getElementById("locate-ffmpeg-picker");

--- a/main.js
+++ b/main.js
@@ -255,6 +255,18 @@ ipcMain.handle("resize-export-window", async (event, width, height) => {
     main_log.info(`export window: new size: ${width}x${height}`);
 });
 
+/**
+ * Attempts to delete render frames
+ */
+ipcMain.handle("clean-up-render-files", () => {
+    try {
+        fsExtra.emptyDirSync(path.resolve(working_dir, "./temp/render")); //clear render cache
+        main_log.info("render cache cleared.");
+    } catch (e) {
+        main_log.error("Couldn't clear all the render cache because some files are still busy.");
+    }
+});
+
 
 
 

